azure-proto-core-test/SkuTests.cs

```diff
@@ -0,0 +1,135 @@
+﻿using azure_proto_core;
+using NUnit.Framework;
+
+namespace azure_proto_core_test
+{
+    class SkuTests
+    {
+        [Test]
+        public void CompareToTests()
+        {
+            Sku sku1 = new Sku();
+            Sku sku2 = null;
+
+            sku1.Name = "name";
```

Comment:
Verify with @markcowl where the constraints are documented for this object?  It is possible this is deserialize only in which case that is "irrelevant"?

Reply:
Pending until getting the exact constraints from the contact. Will update in a separate PR if need to.